extensions/panache/hibernate-orm-panache-common/runtime/src/main/java/io/quarkus/hibernate/orm/panache/common/runtime/CommonPanacheQueryImpl.java

```diff
@@ -79,15 +87,17 @@ private CommonPanacheQueryImpl(CommonPanacheQueryImpl<?> previousQuery, String n
         if (PanacheJpaUtil.isNamedQuery(query)) {
             throw new PanacheQueryException("Unable to perform a projection on a named query");
         }
-
+        if (query != null && query.toLowerCase().startsWith(SELECT)) {
+            throw new PanacheQueryException("Cannot perform projection on a query with an existing select clause: " + query);
+        }
```

Comment:
Can you add the same check on the other projet method ?

Reply:
It's actually already there ;-)

Reply:
OK, cool :)